sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -245,6 +246,55 @@ case class DataSourceResolution(
       catalog: TableCatalog,
       identifier: Identifier,
       create: CreateTableStatement): CreateV2Table = {
+
+    val isCaseSensitive = conf.caseSensitiveAnalysis
+    // Check that columns are not duplicated in the schema
+    val flattenedSchema = SchemaUtils.explodeNestedFieldNames(create.tableSchema)
+    SchemaUtils.checkColumnNameDuplication(
+      flattenedSchema,
+      s"in the table definition of ${identifier.quoted}",
+      isCaseSensitive)
+
+    // Check that columns are not duplicated in the partitioning statement
+    SchemaUtils.checkTransformDuplication(
+      create.partitioning, "in the partition schema", isCaseSensitive)
+
+    // Check that columns are not duplicated in the bucketing specification
+    SchemaUtils.checkTransformDuplication(
+      create.bucketSpec.map(_.asTransform).toSeq, "in the bucket definition", isCaseSensitive)
+    if (create.tableSchema.isEmpty) {
```

Comment:
I don't think creating a table with an empty schema should be allowed.

Reply:
Creating a table may be to simply create a pointer in the metastore over an existing path based table